### PR TITLE
Assorted minor refactors and performance nits

### PR DESCRIPTION
**Pull request by vrmiguel**
I'm familiaring myself with the tembo-cli code and thought it was a good opportunity to suggest these minor changes

### CODE REVIEW

File: tembo-cli/src/cli/context.rs [hunk 1]
```diff
@@ -92,62 +91,43 @@ pub fn tembo_credentials_file_path() -> String {
...
 }
 
 pub fn get_current_context() -> Result<Environment, anyhow::Error> {
     let context = list_context()?;
 
-    let profiles = list_credentail_profiles()?;
+    let profiles = list_credential_profiles()?;
 
-    for mut e in context.environment {
-        if e.set.is_some() && e.set.unwrap() {
```
Comment (vrmiguel) [inline tembo-cli/src/cli/context.rs:118]: @shahadarsh The previous logic does not check the actual value of the `set` variable, is this intended?

File: tembo-cli/src/cli/context.rs [hunk 2]
```diff
@@ -92,62 +91,43 @@ pub fn tembo_credentials_file_path() -> String {
...
-            if e.profile.is_some() {
-                let credential = profiles
-                    .iter()
-                    .filter(|c| &c.name == e.profile.as_ref().unwrap())
-                    .last()
-                    .unwrap();
+    for mut env in context.environment {
+        if let Some(_is_set) = env.set {
+            if let Some(profile) = &env.profile {
+                let credential = profiles.iter().rev().find(|c| &c.name == profile).unwrap();
```
Comment (vrmiguel) [inline tembo-cli/src/cli/context.rs:111]: Same idea as the previous code (iterate and find the last matching element) but starting from the end, doing less work overall

File: tembo-cli/src/cmd/validate.rs [hunk 3]
```diff
@@ -43,7 +43,7 @@ pub fn execute(verbose: bool) -> Result<(), anyhow::Error> {
         let mut file_path = FileUtils::get_current_working_dir();
         file_path.push_str("/tembo.toml");
 
-        let contents = fs::read_to_string(file_path.clone())?;
```
Comment (DarrenBaldwin07) [inline tembo-cli/src/cmd/validate.rs:46]: Awesome!

**Review by DarrenBaldwin07 — APPROVED** [review]
Very clean @vrmiguel

File: tembo-cli/src/cli/context.rs (hunk 1 above)
Comment (shahadarsh) [reply tembo-cli/src/cli/context.rs:118]: Yes, it only sets 1 value at a time so the value for others will be `None`.

File: tembo-cli/src/cli/context.rs [hunk 4]
```diff
@@ -92,62 +91,43 @@ pub fn tembo_credentials_file_path() -> String {
...
+        .map_err(|err| anyhow!("Error reading file {filename}: {err}"))?;
 
-    let credential: Credential = match toml::from_str(&contents) {
-        std::result::Result::Ok(c) => c,
-        Err(e) => {
-            bail!("Issue with format of toml file {filename}: {e}")
-        }
-    };
+    let credential: Credential =
+        toml::from_str(&contents).map_err(|err| anyhow!("Error reading file {filename}: {err}"))?;
```
Comment (shahadarsh) [inline tembo-cli/src/cli/context.rs:130]: This error msg should be `Issue with format of the toml file {filename}` so its more helpful

**Review by shahadarsh — APPROVED** [review]
Nice! Looks good. Just a minor comment on the error msg.

**Pull request merged**